addons/account/models/account_move.py

```diff
@@ -164,6 +165,10 @@ def button_cancel(self):
         for move in self:
             if not move.journal_id.update_posted:
                 raise UserError(_('You cannot modify a posted entry of this journal.\nFirst you should set the journal to allow cancelling entries.'))
+            # We remove all the analytics entries for this journal
+            for line in move.line_ids:
+                if line.analytic_line_ids:
+                    line.analytic_line_ids.unlink()
```

Comment:
Why not `move.mapped('line_ids.analytic_line_ids').unlink()`? That will remove all analytic entries in a single unlink call.